test/extended/util/configv1shim.go

```diff
@@ -0,0 +1,1022 @@
+		return newAPIResourceList(groupVersion, []metav1.APIResource{}, fakeList.APIResources), nil
+	}
+	return realList, err
+}
+
+func (c *ConfigV1DiscoveryClientShim) ServerGroupsAndResources() ([]*metav1.APIGroup, []*metav1.APIResourceList, error) {
+	groups, resources, err := c.configClient.Discovery().ServerGroupsAndResources()
+	if err != nil {
+		return groups, resources, err
+	}
+	if groups != nil {
+		hasConfigGroup := false
+		for i, group := range groups {
+			if group.Name == configGroup {
+				hasConfigGroup = true
+				hasV1Version := false
+				for _, version := range group.Versions {
+					if version.Version == configVersion {
+						hasV1Version = true
+						break
+					}
+				}
+				if !hasV1Version {
+					groups[i].Versions = append(groups[i].Versions, configGroupVersionForDiscovery())
+				}
+				break
+			}
+		}
+		if !hasConfigGroup {
+			groups = append(groups, configAPIGroup())
+		}
+	}
+
+	if resources != nil {
+		fakeList, err := c.fakeClient.Discovery().ServerResourcesForGroupVersion(configGroupVersion)
+		if err != nil {
+			return nil, nil, err
+		}
+		hasConfigGroup := false
+		for i, resource := range resources {
+			if resource.GroupVersion == configGroupVersion {
+				hasConfigGroup = true
+				resources[i].APIResources = mergeResources(fakeList.APIResources, resources[i].APIResources)
+			}
+		}
+		if !hasConfigGroup {
+			resources = append(resources, newAPIResourceList(configGroupVersion, []metav1.APIResource{}, fakeList.APIResources))
+		}
+	}
+
+	return groups, resources, nil
+}
+
+func (c *ConfigV1DiscoveryClientShim) ServerPreferredResources() ([]*metav1.APIResourceList, error) {
+	resources, err := c.configClient.Discovery().ServerPreferredResources()
+	if err != nil {
+		return nil, err
+	}
+
+	fakeList, err := c.fakeClient.Discovery().ServerResourcesForGroupVersion(configGroupVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	hasConfigGroup := false
+	for i, resource := range resources {
+		if resource.GroupVersion == configGroupVersion {
+			hasConfigGroup = true
+			resources[i].APIResources = mergeResources(fakeList.APIResources, resources[i].APIResources)
+		}
+	}
+
+	if !hasConfigGroup {
+		resources = append(resources, newAPIResourceList(configGroupVersion, []metav1.APIResource{}, fakeList.APIResources))
+	}
+
+	return resources, nil
+}
+
+func (c *ConfigV1DiscoveryClientShim) ServerPreferredNamespacedResources() ([]*metav1.APIResourceList, error) {
+	return c.configClient.Discovery().ServerPreferredNamespacedResources()
+}
+
+func (c *ConfigV1DiscoveryClientShim) ServerVersion() (*version.Info, error) {
+	return c.configClient.Discovery().ServerVersion()
+}
+
+func (c *ConfigV1DiscoveryClientShim) OpenAPISchema() (*openapi_v2.Document, error) {
+	// TODO(jchaloup): once needed implement this method as well
+	panic(fmt.Errorf("APIServer not implemented"))
+}
+
+func (c *ConfigV1DiscoveryClientShim) OpenAPIV3() openapi.Client {
+	return c.configClient.Discovery().OpenAPIV3()
```

Comment:
not sure if we shouldn't panic in these last 3 as well

Reply:
Done